src/fraction/unicode_fromto_str.rs

```diff
@@ -0,0 +1,589 @@
+    /// This will look OK without font support
+    /// ```
+    /// use fraction::Fraction;
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(1u8,2u8).display_supsub_mixed()),
+    ///   "¹/₂"
+    /// );
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(3u8,2u8).display_supsub_mixed()),
+    ///   "1¹/₂"
+    /// );
+    /// ```
+    pub fn display_supsub_mixed(&self) -> impl Display + '_ {
+        struct S<'a, T: Clone + Integer + fmt::Display>(&'a GenericFraction<T>);
+        impl<'a, T> fmt::Display for S<'a, T>
+        where
+            T: Clone + Integer + fmt::Display + From<u8>,
+        {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                match &self.0 {
+                    GenericFraction::Rational(s, r) if r.fract() != *r && !r.denom().is_one() => {
+                        write!(
+                            f,
+                            "{}{}{}",
+                            s,
+                            r.trunc().numer(),
+                            GenericFraction::Rational(Sign::Plus, r.fract().clone())
+                                .display_supsub()
+                        )
+                    }
+                    _ => write!(f, "{}", self.0.display_supsub()),
+                }
+            }
+        }
+        S(self)
+    }
+
+    /// Parse a unicode string
+    /// The string can be:
+    /// - A normal fraction e.g. "1/2"
+    /// - A vulgar fraction e.g. "½"
+    /// - ~A mixed vulgar fraction "1½"~
+    /// - A unicode fraction e.g. "1⁄2" where '⁄' can be any of:
+    ///   - '/' ASCII SOLIDUS
+    ///   - '⁄' FRACTION SLASH
+    ///   - '∕' DIVISION SLASH
+    ///   - '÷' DIVISION SIGN
+    /// - A mixed unicode fraction: "1\u{2063}1⁄2": 1⁤1⁄2
+    ///   - '\u{2064}' INVISIBLE PLUS
+    ///   - '\u{2063}' INVISIBLE SEPARATOR
+    ///   - NOT ~'\u{2062}' INVISIBLE TIMES~
+    /// - A super-subscript fraction "¹/₂"
+    /// - A mixed super-subscript fraction  "1¹/₂"
+    ///
+    /// Focus is on being lenient towards input rather than being fast.
+    /// ```
+    /// use fraction::Fraction;
+    /// let v = vec![
+    ///  ("1/2", Fraction::new(1u8,2u8)),
+    ///  ("-1/2", Fraction::new_neg(1u8,2u8)),
+    ///  ("½", Fraction::new(1u8,2u8)),
+    ///  // ("1½", Fraction::new(1u8,2u8)),
+    ///  // ("-1½", Fraction::new_neg(1u8,2u8)),
+    ///  ("1⁄2", Fraction::new(1u8,2u8)),
+    ///  ("-1⁄2", Fraction::new_neg(1u8,2u8)),
+    ///  ("1⁤1⁄2", Fraction::new(3u8,2u8)),
+    ///  ("-1⁤1⁄2", Fraction::new_neg(3u8,2u8)),
+    ///  ("¹/₂", Fraction::new(1u8,2u8)),
+    ///  ("-¹/₂", Fraction::new_neg(1u8,2u8)),
+    ///  ("1¹/₂", Fraction::new(3u8,2u8)),
+    ///  ("-1¹/₂", Fraction::new_neg(3u8,2u8)),
+    /// ];
+    /// for (f_str, f) in v {
+    ///   assert_eq!(Fraction::parse_unicode(f_str), Ok(f))
+    /// }
+    /// ```
+    pub fn parse_unicode(input: &str) -> Result<Self, ParseError> {
```

Comment:
For consistency, could we call this method `from_unicode_str`?

Reply:
Yes can do. I also realized it does a bit weird things for certain strings, bc I did starts_with for nan and infinity, so those actually parse (see BOGUS tests)